src/movies.js

```diff
@@ -1,16 +1,123 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+    const directorsList = movies.map(function(name) {
+            return name.director
+        })
+    let uniqueDirectorsList = []
+    for (let director of directorsList) {
+        if (uniqueDirectorsList.indexOf(director) === -1) {
+            uniqueDirectorsList.push(director)
+        }
+    }
+  return uniqueDirectorsList
+    }
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies) {
+    const listOfSpielbergDramas = movies.filter(function (movie) {
+        if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')) {
+          return true
+        }
```

Comment:
With filter, you only need a return statement with your filtering logic.
movies.filter(function (movie) {
        return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')
        })
Filter will return all the element that match movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')